### PR TITLE
Fix DB write out for coupling

### PR DESCRIPTION
**Pull request by albeanth**
## Description
This PR fixes a bug for how databases were written when using tight coupling. Previously, databases were written out on the `interactEveryNode` and then never written out/updated after the coupling iterations. So in effect, the updated values were never written to the DB. This PR resolves this by doing nothing on database3.py::Database::interactEveryNode if tight coupling is requested. Then only after the tight coupling iterations are complete is the database written out. 

---

## Checklist

- [x] This PR has only one purpose or idea.
- [x] Tests have been added/updated to verify that the new/changed code works.

- [x] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [x] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [x] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [x] The documentation is still up-to-date in the `doc` folder.
- [x] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

File: armi/bookkeeping/db/databaseInterface.py [hunk 1]
```diff
@@ -145,7 +145,12 @@ def interactEveryNode(self, cycle, node):
 
         DBs should receive the state information of the run at each node.
         """
-        # skip writing for last burn step since it will be written at interact EOC
```
Comment (onufer) [inline armi/bookkeeping/db/databaseInterface.py:148]: please copy this comment over

**Review by onufer — CHANGES_REQUESTED** [review]

File: armi/bookkeeping/db/databaseInterface.py [hunk 2]
```diff
@@ -145,7 +145,12 @@ def interactEveryNode(self, cycle, node):
...
+        if self.o.cs["numCoupledIterations"]:
```
Comment (onufer) [inline armi/bookkeeping/db/databaseInterface.py:148]: add comment saying that h5 cant handle writing more than once so we only write at the end

Comment (albeanth) [reply armi/bookkeeping/db/databaseInterface.py:153]: Addressed https://github.com/terrapower/armi/pull/1005/commits/661e26a9a2a148719549ff717ce4fd0b745b2ef8

File: armi/bookkeeping/db/databaseInterface.py (hunk 1 above)
Comment (albeanth) [reply armi/bookkeeping/db/databaseInterface.py:148]: Addressed https://github.com/terrapower/armi/pull/1005/commits/661e26a9a2a148719549ff717ce4fd0b745b2ef8

**Review by onufer — APPROVED** [review]

**Comment by john-science** [comment]
You can see [here](https://coveralls.io/builds/54797142/source?filename=armi%2Foperators%2Foperator.py#L384) that two of your new lines of code aren't covered by your unit tests.  Thus the coverage failure.

**Review by john-science — APPROVED** [review]

**Pull request merged**